mockrxandroidble/src/main/java/com/polidea/rxandroidble2/mockrxandroidble/RxBleScanRecordMock.java

```diff
@@ -0,0 +1,185 @@
+    private SparseArray<byte[]> manufacturerSpecificData;
+    private Map<ParcelUuid, byte[]> serviceData;
+    private int txPowerLevel;
+    private String deviceName;
+    private byte[] bytes;
+
+    public RxBleScanRecordMock(
+        int advertiseFlags,
+        List<ParcelUuid> serviceUuids,
+        SparseArray<byte[]> manufacturerSpecificData,
+        Map<ParcelUuid, byte[]> serviceData,
+        int txPowerLevel,
+        String deviceName,
+        byte[] bytes) {
+        this.advertiseFlags = advertiseFlags;
+        this.serviceUuids = serviceUuids;
+        this.manufacturerSpecificData = manufacturerSpecificData;
+        this.serviceData = serviceData;
+        this.txPowerLevel = txPowerLevel;
+        this.deviceName = deviceName;
+        this.bytes = bytes;
+    }
+
+    /**
+     * Builder class for {@link RxBleScanRecordMock}.
+     */
+    public static class Builder {
+        private int advertiseFlags;
+        private List<ParcelUuid> serviceUuids;
+        private SparseArray<byte[]> manufacturerSpecificData;
+        private Map<ParcelUuid, byte[]> serviceData;
+        private int txPowerLevel;
+        private String deviceName;
+        private byte[] bytes;
+
+        public Builder() {
+            serviceUuids = new ArrayList<ParcelUuid>();
+            manufacturerSpecificData = new SparseArray<>();
+            serviceData = new HashMap<>();
+        }
+
+        /**
+         * Set the advertise flags
+         */
+        public Builder setAdvertiseFlags(int advertiseFlags) {
+            this.advertiseFlags = advertiseFlags;
+            return this;
+        }
+
+        /**
+         * Add a service UUID
+         */
+        public Builder addServiceUuid(ParcelUuid uuid) {
+            serviceUuids.add(uuid);
+            return this;
+        }
+
+        /**
+         * Add manufacturer specific data
+         */
+        public Builder addManufacturerSpecificData(int manufacturerId, byte[] manufacturerSpecificData) {
+            this.manufacturerSpecificData.append(manufacturerId, manufacturerSpecificData);
+            return this;
+        }
+
+        /**
+         * Add service data
+         */
+        public Builder addServiceData(ParcelUuid uuid, byte[] data) {
+            serviceData.put(uuid, data);
+            return this;
+        }
+
+        /**
+         * Set the tx power level
+         */
+        public Builder setTxPowerLevel(int txPowerLevel) {
+            this.txPowerLevel = txPowerLevel;
+            return this;
+        }
+
+        /**
+         * Set the device name
+         */
+        public Builder setDeviceName(String deviceName) {
+            this.deviceName = deviceName;
+            return this;
+        }
+
+        /**
+         * Build {@link RxBleScanRecordMock}.
+         *
+         * @throws IllegalArgumentException If the scan record cannot be built.
+         */
+        public RxBleScanRecordMock build() {
+            // TODO: compile to bytes
```

Comment:
How about throwing an exception when trying to get bytes from RxBleScanRecordMock? Wouldn't it be clearer then that it is not implemented?